src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -396,10 +399,23 @@ internal override async ValueTask ShutdownWriteCompleted(CancellationToken cance
             await _state.ShutdownWriteCompletionSource.Task.ConfigureAwait(false);
         }
 
+        internal override async ValueTask ShutdownCompleted(CancellationToken cancellationToken = default)
+        {
+            ThrowIfDisposed();
+
+            using CancellationTokenRegistration registration = cancellationToken.Register(() =>
+                _state.ShutdownCompletionSource.SetException(new OperationCanceledException("Shutdown was canceled", cancellationToken))
```

Comment:
You should use `ExceptionDispatchInfo.SetCurrentStackTrace` on the `OperationCanceledException`. Otherwise the caller will not get a full stack trace.